### DISCUSSION TITLE
quarkus:dev not working in 2.8.2.Final

### DISCUSSION THREAD

**Discussion by SvenMarien** [quarkusio/quarkus#25205] [Q&A]
Befor submitting an official bug or issue I would like to know if someone else is experiencing the same exception as I have. After upgrading quarkus to version 2.8.2.Final I get the following exception when starting mvn quarkus:dev

[ERROR] Failed to execute goal io.quarkus.platform:quarkus-maven-plugin:2.8.2.Final:dev
Caused by: java.lang.NullPointerException
    at io.quarkus.maven.DevMojo.addProject (DevMojo.java:741)
    at io.quarkus.maven.DevMojo.newLauncher (DevMojo.java:1028)
    at io.quarkus.maven.DevMojo$DevModeRunner.<init> (DevMojo.java:858)

Looking at the code of DevMojo it is due to this variable being null and called upon on line 741:
final ArtifactSources sources = module.getSources();

Is this due to one of my maven modules not having sources or resources? Bit stuck here...

[the remark this answers is not in the document]
**Reply by aloubyansky** [reply]
I'll fix it. Thanks @SvenMarien

[the remark this answers is not in the document]
**Reply by aloubyansky** [reply]
@SvenMarien I'm trying to create a reproducer, could you please provide more details about this module that has not sources? Which packaging type does it have? pom/jar? Is it the main app module or a dependency? If you have a reproducer I could use that would be even better. Thanks.

[the remark this answers is not in the document]
**Reply by aloubyansky** [reply]
I couldn't reproduce it so far. I can surely fix it but I want to add test for it, for which I need a reproducer. I'll need your help here @SvenMarien Tell me more, please, about that module.

[the remark this answers is not in the document]
**Reply by SvenMarien** [reply]
hey, thanks for the swift response.
Thing is that I do not know on which module it generates the nullpointerexception. I got a classic maven project directory structure:
subpersub
 - subpersub-api
 - subpersub-business
 - subpersub-client
 - subpersub-domain
 - subpersub-flyway
 and the exception doesn't tell me on which module it bugs out.

[the remark this answers is not in the document]
**Reply by SvenMarien** [reply]
Reverting to 2.8.0.CR1 deblocks my issue though. So there has to be a change there between the CR1 and Final that generates the NPE.  Perhaps adding a null-check on the ArtifactSources variable could help? Or is the addProject() called for a non-existing module or for the parent module (which does not need to be done)? Too bad I can't debug the plugin.
Any idea how I could make a reproducer you would like to have? Can't zip our project to you I guess... :)

[the remark this answers is not in the document]
**Reply by aloubyansky** [reply]
Looking at the stacktrace I can make it go away, sure. But I wanted to add a test to make sure we don't break it in the future.
There are a few options to provide a reproducer. If you can you could push it to github, that would be the easiest. You could also attach a ZIP to a github issue. Or you could also send it to me by email, if you like. Let me know your preference. Thanks!

[the remark this answers is not in the document]
**Reply by SvenMarien** [reply]
Hmm, it's a commercial project of one of my clients. Can't give you the complete sources, but let me see if I can strip it down enough so I can zip it to you.

[the remark this answers is not in the document]
**Reply by SvenMarien** [reply]
I have a zip file which reproduces the problem. Will make a gitlab issue of this and attach the zip there, if that's ok for you?

[the remark this answers is not in the document]
**Reply by SvenMarien** [reply]
https://github.com/quarkusio/quarkus/issues/25255

[the remark this answers is not in the document]
**Reply by aloubyansky** [reply]
Absolutely. Thanks @SvenMarien I'll have a look.

[the remark this answers is not in the document]
**Reply by aloubyansky** [reply]
That was super useful @SvenMarien. The bug was in processing the maven-jar-plugin config. https://github.com/quarkusio/quarkus/pull/25266 will fix it. Thanks!
